DuckDuckGo/SubscriptionDebugViewController.swift

```diff
@@ -176,19 +174,46 @@ final class SubscriptionDebugViewController: UITableViewController {
             default: break
             }
         case .environment:
-            switch EnvironmentRows(rawValue: indexPath.row) {
-            case .staging: setEnvironment(.staging)
-            case .production: setEnvironment(.production)
-            default: break
-            }
+            guard let subEnv: EnvironmentRows = EnvironmentRows(rawValue: indexPath.row) else { return }
+            changeSubscriptionEnvironment(envRows: subEnv)
         case .none:
             break
         }
-
         tableView.deselectRow(at: indexPath, animated: true)
     }
-    // swiftlint:enable cyclomatic_complexity
     
+    private func changeSubscriptionEnvironment(envRows: EnvironmentRows) {
+        var subEnvDesc: String
+        switch envRows {
+        case .staging:
+            subEnvDesc = "STAGING"
+        case .production:
+            subEnvDesc = "PRODUCTION"
+        }
+        let message = """
+                    Are you sure you want to change the purchase platform to \(subEnvDesc)?
```

Comment:
This is not `purchase platform` but service environment.
And tbt the more I think about it the less I think the environment switch on iOS makes sense - the type of environment (for subscriptions at least) is dependant on app bundle identifier (so on the target) since the corresponding app entry in App Store Connect defines which BE is "connected" to the purchases hence what BE endpoints should the app call.
You can leave it for now but definitely something I have on my mind to clear up. If you have ideas or if anything is unclear please ping me, I can elaborate.

Reply:
I didn't know that, lol I could have saved some time XD
Message fixed.